sys/kern/subr_bus.c

```diff
@@ -1240,6 +1245,9 @@ devclass_alloc_unit(devclass_t dc, device_t dev, int *unitp)
 		}
 	}
 
+	if (__predict_false(unit >= DEVICE_UNIT_MAX))
+		return (ERANGE);
```

Comment:
EINVAL is better, plus this needs to be added to the docs here and in this routine's callers.